libraries/psibase_http/handle_request.cpp

```diff
@@ -647,6 +647,12 @@ namespace psibase::http
 
             auto        startTime = steady_clock::now();
             HttpRequest data;
+            for (auto iter = req.begin(); iter != req.end(); ++iter)
+            {
+               data.headers.emplace_back(
+                   HttpHeader{std::string(iter->name_string()), std::string(iter->value())});
```

Comment:
Should I just whitelist the request headers that are checked?

> may reveal secrets to services

It should only reveal secrets to the default or registered http server services, though, right? Do you mean it may reveal secrets to node operators?

Reply:
I'm mainly thinking of secrets that are shared between the client and the node operator. At the moment there's only the native login token, which is only for `x-admin` and therefore explicitly excluded from wasm.  Any tokens involved in billing for queries could become a problem in the future though.

Reply:
Also, proxies might not always remove authentication headers.

Reply:
I am thinking that we want to make this as similar to using your own webserver as possible, right? Whitelisting therefore feels too restrictive to me.

I think maybe it makes more sense to blacklist the particular headers that we don't want to expose to wasm.

If so, I am thinking the exclude list is just the authorization headers:
* `Authorization`
* `Proxy-Authorization`

Do you agree with this?

Reply:
I agree that whitelisting feels too restrictive. OTOH, it's all too easy for blacklisting to miss something accidentally. I don't think there's a perfect solution, here. I just want to make sure that the issue has been considered.

Reply:
I decided to use blacklisting because I value being able to expect a header to pass through by default